jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+import os
+
+class FileSystemCache:
+
+  def __init__(self, path: str):
+    """Sets up a cache at 'path'. Cached values may already be present."""
+    if not os.path.isdir(path):
+      os.mkdir(path)
+    self._path  = path
+
+  def get(self, key: str) -> Optional[bytes]:
+    """Returns None if 'key' isn't present."""
+    if key == "":
```

Comment:
Ah I made a mistake earlier. This should actually be `if not key`, see https://google.github.io/styleguide/pyguide.html#2144-decision